docs/reST/ext/indexer.py

```diff
@@ -57,21 +55,31 @@ def prep_document_info(app, env, docname):
     except AttributeError:
         pass
     else:
-        to_remove = [k for k, v in descinfo_tbl.items()
-                     if v['docname'] == docname]
+        to_remove = [k for k, v in descinfo_tbl.items() if v["docname"] == docname]
         for k in to_remove:
             del descinfo_tbl[k]
 
 
 def collect_document_info(app, doctree):
     doctree.walkabout(CollectInfo(app, doctree))
 
+
 class CollectInfo(Visitor):
 
     """Records the information for a document"""
-    
-    desctypes = set(['data', 'function', 'exception', 'class',
-                     'attribute', 'method', 'staticmethod', 'classmethod'])
+
+    desctypes = set(
+        [
```

Comment:
Minor nitpick here
Why declare a list and make it a set, when you can directly declare it as a set?

Reply:
Good catch.
I didn't even know the `dict`-esque syntax to do that existed.